### PR TITLE
add frozen message

### PR DESCRIPTION
**Pull request by FKD13**

### CODE REVIEW

File: src/tasks/fetcher.py [hunk 1]
```diff
@@ -69,12 +69,17 @@ async def fetch(self):
...
                     # Filter laps by freeze time
                     if self._settings.site.freeze is not None:
-                        laps: list[Lap] = [
+                        new_laps: list[Lap] = [
                             lap
                             for lap in laps
                             if lap.timestamp <= self._settings.site.freeze
                         ]
 
+                        # If the filter removed laps, we now the scoreboard is frozen
```
Comment (Topvennie) [inline src/tasks/fetcher.py:78]: kkkkkkkkkkkkkkkkkkkkkkkk

File: static/admin.js [hunk 2]
```diff
@@ -17,11 +17,15 @@ websocket.onopen = (_event) => {
...
-        setMessage(data.data);
+    if (!('ping' in data)) {
+        console.log(data)
+        if (data['topic'] === 'counts') {
+            setData(data.data);
+        } else if (data['topic'] === 'message') {
+            setMessage(data.data);
+        } else if (data['topic'] === 'frozen') {
+            setFrozen(data.data)
+        }
```
Comment (Topvennie) [inline static/admin.js:28]: Kris would say to avoid the if else stack by using a dictionary with functions in its values

File: static/admin.js [hunk 3]
```diff
@@ -72,13 +76,15 @@ let message_handlers = {
 
 admin_feed.onmessage = e => {
     let data = JSON.parse(e.data)
-    console.log(data)
     if ('topic' in data && 'data' in data) {
+        console.log(data)
```
Comment (Topvennie) [inline static/admin.js:80]: Do we still want this?

**Review by Topvennie — APPROVED** [review]

File: static/admin.js (hunk 2 above)
Comment (FKD13) [reply static/admin.js:28]: I know like I did with the admin messages, but this is the last one I swear

File: static/admin.js (hunk 3 above)
Comment (FKD13) [reply static/admin.js:80]: Can be useful, but the Firefox web socket support has got better. I would keep it for now, it's only on the admin page anyway

File: src/tasks/fetcher.py (hunk 1 above)
Comment (FKD13) [reply src/tasks/fetcher.py:78]: A bit of a hack indeed, I did not want to create a new thread just to check this :sweat_smile:

Comment (Topvennie) [reply src/tasks/fetcher.py:78]: I meant `now` -> `know` :upside_down_face:

Comment (FKD13) [reply src/tasks/fetcher.py:78]: ah whelp

File: static/admin.js (hunk 2 above)
Comment (Topvennie) [reply static/admin.js:28]: That's what they all say

**Pull request merged**